kbase-extension/static/kbase/js/widgets/narrative_core/kbaseDataCard.js

```diff
@@ -1,215 +1,202 @@
+ *      type: str, the shortened type of object. E.g., if it's KBaseGenomes.Genome-2.1, then "Genome"
+ *      viewType: str, the viewed object type. Mainly for Genomes that should have the scientific name appended.
  *
-*/
+ *      //optional values
+ *      moreContent: jquery object,
+ *      is_set: boolean, true if this is a Set object,
+ *      max_name_length: int, overrides the Config'd max_name_length if present (chops down the
+ *                       objects name to some maximum number of characters)
+ *  });
+ */
+
+define ([
+    'bootstrap',
+    'util/icon',
+    'bluebird',
+    'util/bootstrapDialog',
+    'util/timeFormat',
+    'kbase/js/widgets/narrative_core/kbaseCardLayout',
+    'narrativeConfig',
+    'jquery'
+], function(
+    bootstrap,
+    Icon,
+    Promise,
+    BootstrapDialog,
+    TimeFormat,
+    kbaseCardLayout,
+    Config,
+    $
+) {
+    'use strict';
+    function KbaseDataCard(entry) {
+        var self = this,
+            object_info = entry.object_info,
+            maxNameLength = Config.get('data_panel').max_name_length;
+        if (entry.max_name_length) {
+            maxNameLength = entry.max_name_length;
+        }
 
-define (
-    [
-        'bootstrap',
-        'util/icon',
-        'bluebird',
-        'util/bootstrapDialog',
-        'util/timeFormat',
-        'kbase/js/widgets/narrative_core/kbaseCardLayout',
-        'narrativeConfig',
-        'jquery'
-    ], function(
-        bootstrap,
-        Icon,
-        Promise,
-        BootstrapDialog,
-        TimeFormat,
-        kbaseCardLayout,
-        Config,
-        $
-    ) {
-        function KbaseDataCard(entry) {
-            var self = this;
-            var object_info = entry.object_info;
-            
-            //params
-            var shortName = entry.name ? entry.name : object_info[1];
-            var version = entry.version ? entry.version : ('v' + object_info[4]);
-            var date = entry.date ? entry.date : TimeFormat.getTimeStampStr(object_info[3]);
-            var type = entry.type;
-            var editBy = entry.editedBy ? entry.editedBy : (' by ' + object_info[5]);
-            if(!entry.type){
-                var type_tokens = object_info[2].split('.');
-                type = entry.type ? entry.type : type_tokens[1].split('-')[0];
-            }
+        //params
+        var shortName = entry.name ? entry.name : object_info[1],
+            version = entry.version ? entry.version : ('v' + object_info[4]),
+            date = entry.date ? entry.date : TimeFormat.getTimeStampStr(object_info[3]),
+            editBy = entry.editedBy ? entry.editedBy : (' by ' + object_info[5]);
+
+        // in order - entry.viewType > entry.type > parsing it out of the object_info type string.
+        var objectType = entry.type ? entry.type : object_info[2].split('.')[1].split('-')[0],
+            viewType = entry.viewType ? entry.viewType : entry.type;
+        viewType = viewType ? viewType : objectType;
```

Comment:
viewType set twice seems pretty confusing. Maybe break it out into if statements?

Reply:
You're right. Got too clever for my own good. Fixed.